bigtable_rs/src/bigtable/read_rows.rs

```diff
@@ -33,6 +35,35 @@ pub async fn decode_read_rows_response(
     Ok(rows)
 }
 
+/// Decode the stream of `ReadRowsResponse` into a stream of `Result<(RowKey, Vec<RowCell>)>>`.
+pub async fn decode_read_rows_response_stream<'a>(
+    timeout: &'a Option<Duration>,
+    mut rrr: Streaming<ReadRowsResponse>,
+) -> Pin<Box<dyn Stream<Item = Result<(RowKey, Vec<RowCell>)>> + Send + 'a>> {
+    let started = Instant::now();
+
+    Box::pin(async_stream::stream! {
+        while let Ok(message) = rrr.message().await {
+            match message {
+                Some(response) => {
+                    if let Some(timeout) = timeout.as_ref() {
+                        if Instant::now().duration_since(started) > *timeout {
+                            yield Err(Error::TimeoutError(timeout.as_secs()));
+                            break;
+                        }
+                    }
+
+                    let rows_part = decode_read_rows_response_to_vec(response.chunks);
+                    for row in rows_part {
+                        yield row;
+                    }
```

Comment:
This "for" loop essentially unpacks again of the existing stream packages from the received message. Do you think it is necessary? Perhaps it is better if we do not do another level of unpacking and return `Vec<Result<(RowKey, Vec<RowCell>)>>` as a chunk of stream here directly?

Reply:
This seems reasonable! Back pressure can only be applied at the row level anyways.

Reply:
Ok, so I played with this and things got a bit ugly and the ergonomics for the caller felt really off. Can you take a look at the latest implementation? Without `Box<Pin<T>>` it seems a lot nicer and I'm not sure there's much of a performance difference.

Reply:
Aha, so it is unpacked but with a flat_map operation. I think it is good enough and also as you said easier to consume as well. I think it looks good, thanks for updating :)